### DISCUSSION TITLE
RP2040 + ILI9341 + SPI How much time should clearing screen take  ?

### DISCUSSION THREAD

**Discussion by ValouBambou** [Bodmer/TFT_eSPI#3210] [Raspberry Pi Pico]
Hi, I'm building an application using a Raspberry Pi Pico and an ili9341 320x240 display connected with SPI. My problem is that the display is slow (85 ms to clear the screen). I read about how SPI is slow but since I'm using some a custom library I'm not sure if my slowness is due to software issue (my lib is poorly written) or hardware SPI issue. The only number I can find in the README is the time for clearing a bigger screen with parallel interface, which is supposed to be faster than SPI. So using TFT_eSPI what time should I expect to take for a screen cleared in this context ?

**Comment by CoreyCoop** [comment]
I'm using a 320x240 ili9341 display, with the SPI frequency set to 70,000,000, and a fill page clear takes 56MS.  It was a little faster at 80M, but there was a not that that was marginal, so I went with the 70M speed.

**Comment by ValouBambou** [comment]
What library are you using for 56 ms ? This one ? Also, what optimization level have you used when compiling ? 

I managed to reach 25 ms for my custom driver (from the 85 ms, with SPI frequency of 64,000,000 in both cases) with specialization at the cost of generality, but I was expecting something around 10 or 15 ms at best. This goal is still quite far away, but I'm not sure if it is reachable.

**Comment by CoreyCoop** [comment]
Of course it's this library!  I'm using no optimization, still in debug phase, and I'm using the standard driver for the RP2040.

**Comment by ValouBambou** [comment]
That's nice in debug mode. I will be interested to see what time you get when compiling with a more aggressive optimization level like -O2 or -O3.
